paddle/fluid/framework/ir/mkldnn/conv_activation_mkldnn_fuse_pass.cc

```diff
@@ -272,73 +163,114 @@ Conv2DGeluFusePass::Conv2DGeluFusePass() {
-REGISTER_PASS_CAPABILITY(conv_hard_swish_mkldnn_fuse_pass)
-    .AddCombination(
-        paddle::framework::compatible::OpVersionComparatorCombination()
-            .LE("conv2d", 1)
-            .EQ("hard_swish", 0));
+    if (!IsCompat(subgraph, g)) {
+      LOG(WARNING) << "conv_activation_mkldnn_fuse_pass op compat failed.";
+      return;
+    }
 
-REGISTER_PASS(conv_mish_mkldnn_fuse_pass,
-              paddle::framework::ir::Conv2DMishFusePass);
-REGISTER_PASS_CAPABILITY(conv_mish_mkldnn_fuse_pass)
-    .AddCombination(
-        paddle::framework::compatible::OpVersionComparatorCombination()
-            .LE("conv2d", 1)
-            .EQ("mish", 1));
+    GET_IR_NODE_FROM_SUBGRAPH(conv_weight, conv_weight, conv_act_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(conv_out, conv_out, conv_act_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(conv, conv, conv_act_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(activation_out, activation_out, conv_act_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(activation, activation, conv_act_pattern);
 
-REGISTER_PASS(conv_hard_sigmoid_mkldnn_fuse_pass,
-              paddle::framework::ir::Conv2DHardSigmoidFusePass);
-REGISTER_PASS_CAPABILITY(conv_hard_sigmoid_mkldnn_fuse_pass)
-    .AddCombination(
-        paddle::framework::compatible::OpVersionComparatorCombination()
-            .LE("conv2d", 1)
-            .EQ("hard_sigmoid", 0));
+    OpDesc* conv_op = conv->Op();
+    conv_op->SetOutput("Output", {activation_out->Name()});
+
+    if (act_type == "gelu" && activation->Op()->HasAttr("approximate")) {
+      act_type = BOOST_GET_CONST(bool, activation->Op()->GetAttr("approximate"))
+                     ? "gelu_tanh"
+                     : "gelu_erf";
+    }
+
+    conv_op->SetAttr("fuse_activation", act_type);
 
-REGISTER_PASS(conv_gelu_mkldnn_fuse_pass,
-              paddle::framework::ir::Conv2DGeluFusePass);
-REGISTER_PASS_CAPABILITY(conv_gelu_mkldnn_fuse_pass)
+    // MKLDNN ops use alpha and beta as activation parameters but paddle ops are
+    // not generalized
+    if (act_type == "relu6") {
+      conv_op->SetAttr(
+          "fuse_alpha",
+          BOOST_GET_CONST(float, activation->Op()->GetAttr("threshold")));
+    } else if (act_type == "hard_sigmoid") {
+      conv_op->SetAttr("fuse_alpha",
+                       activation->Op()->GetAttrIfExists<float>("slope"));
+      conv_op->SetAttr("fuse_beta",
+                       activation->Op()->GetAttrIfExists<float>("offset"));
+    } else if (act_type == "swish") {
+      conv_op->SetAttr("fuse_alpha",
+                       activation->Op()->GetAttrIfExists<float>("beta"));
+      conv_op->SetAttr("fuse_beta",
+                       activation->Op()->GetAttrIfExists<float>("alpha"));
+    } else {
+      conv_op->SetAttr("fuse_alpha",
+                       activation->Op()->GetAttrIfExists<float>("alpha"));
+      conv_op->SetAttr("fuse_beta",
+                       activation->Op()->GetAttrIfExists<float>("beta"));
+    }
```

Comment:
I don't like a solution with else if ladder.
Please use a map from: paddle/fluid/framework/ir/mkldnn/elt_act_mkldnn_fuse_pass.cc:37
This map is cleaner IMO.

Reply:
Maybe we should make a common pass for fusing op+activation or make it unified in some another way since the map that you have made in elt+act can be reused basically everywhere where activation is being fused into preceding op

Reply:
This is very good idea. I wonder if we should include `sqrt`, `abs` or `clip` because they use same mechanism, but they're not activations. Maybe just rename "operator_activation_pass" into something more inclusive?

Reply:
You can add it, why not.
I'm afraid that currently we don't have a better name for it.
If you will get it I think it can be renamed.

Reply:
OneDNN is describing `sqrt`, `abs` and activations like `relu`, `gelu` etc. as `eltwise` operations, but in PaddlePaddle elementwise operations are totally different thing that in oneDNN, so I don't think that there exist a better option than just treating it as `activation`